extensions/measurement-tracking/src/panels/PanelMeasurementTableTracking/index.js

```diff
@@ -118,27 +115,47 @@ function PanelMeasurementTableTracking({ servicesManager, extensionManager }) {
   };
 
   const createReport = async () => {
-    const measurements = MeasurementService.getMeasurements();
-    const trackedMeasurements = measurements.filter(
-      m =>
-        trackedStudy === m.referenceStudyUID &&
-        trackedSeries.includes(m.referenceSeriesUID)
-    );
+    const loadingDialogId = UIDialogService.create({
+      showOverlay: true,
+      isDraggable: false,
+      centralize: true,
+      // TODO: Create a loading indicator component + zeplin design?
+      content: () => <div className="text-primary-active">Loading...</div>
+    });
 
-    const dataSources = extensionManager.getDataSources();
-    // TODO -> Eventually deal with multiple dataSources.
-    // Would need some way of saying which one is the "push" dataSource
-    const dataSource = dataSources[0];
-
-    DICOMSR.storeMeasurements(
-      trackedMeasurements,
-      dataSource,
-      naturalizedReport => {
-        DisplaySetService.makeDisplaySets([naturalizedReport], {
-          madeInClient: true,
-        });
-      }
-    );
+    try {
+      const measurements = MeasurementService.getMeasurements();
+      const trackedMeasurements = measurements.filter(
+        m =>
+          trackedStudy === m.referenceStudyUID &&
+          trackedSeries.includes(m.referenceSeriesUID)
+      );
+
+      const dataSources = extensionManager.getDataSources();
+      // TODO -> Eventually deal with multiple dataSources.
+      // Would need some way of saying which one is the "push" dataSource
+      const dataSource = dataSources[0];
+
+      const { message } = await DICOMSR.storeMeasurements(
+        trackedMeasurements,
+        dataSource,
+        naturalizedReport => {
+          DisplaySetService.makeDisplaySets([naturalizedReport], {
+            madeInClient: true,
+          });
+        }
+      );
+
+      UINotificationService.show({ title: 'STOW SR', message, type: 'success' });
+    } catch (error) {
+      UINotificationService.show({
+        title: 'STOW SR',
+        message: error.message || 'Failed to store measurements',
+        type: 'error',
+      });
```

Comment:
This could technically show if the `DisplaySetService` fails, but the `storeMeasurements` call is successful. I'm not sure of the best way to "track progress" and identify the point of failure to make sure this is accurate; especially given that `makeDisplaySets` is in a callback for the thing we primarily care about.

Reply:
Updated. Thx.